src/api/config.c

```diff
@@ -612,6 +612,49 @@ m64p_error ConfigShutdown(void)
 /* Selector functions, exported outside of the Core */
 /* ------------------------------------------------ */
 
+EXPORT m64p_error CALL ConfigExternalGetParameter(const char *FileName, const char *SectionName, const char *ParamName, char* ParamPtr, int ParamMaxLength)
+{
+    FILE *fPtr;
+    char buffer[ParamMaxLength + 64]; //ParamMaxLength is the size of the value, we add 64 to make sure we have enough room to store the rest of the line.
```

Comment:
> I must not understand C at all, doesn't the variable get destroyed automatically when the function ends?

Local variables are allocated on the stack (or in a register) at the beginning of the function and de-allocated at the end of the function, this is done at compile time. http://www.avabodh.com/cin/memorymanagement.html

The problem here is that the compiler doesn't know the value of ParamMaxLength at compile time, that's why you have to use dynamic allocation malloc+free. I'm actually surprised the the compiler didn't complained about this.

In this situation I would rather use a fixed length for buffer, being the max size of a line, maybe 4096 like we discussed before?